### ISSUE TITLE
domain neutral instrumentation fails in 1.17.0

### ISSUE THREAD

**Issue by scownTownV** [DataDog/dd-trace-dotnet#735]
**Describe the bug**
After upgrading from 1.16.2 to 1.17.0 per this [issue](https://github.com/DataDog/dd-trace-dotnet/issues/726), and re-enabling domain neutral instrumentaton, the tracing agent logs a warn: `Method replacement found but the managed profiler has not yet been loaded into AppDomain` and fails to capture traffic from domain neutral assemblies.

**To Reproduce**

1. From .NET tracer version 1.16.2 Install 1.17.0
2. Add DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION as a System environment variable with value true
3. Restart machine

**Expected behavior**
domain neutral assemblies are captured in DD trace views

**Runtime environment (please complete the following information):**
- Instrumentation mode: Automatic
- Tracer version: 1.17.0
- OS: Windows Server 2016 Datacenter
- CLR: .NET Framework 4.7.2

**Additional context**
- Was working fine in tracer version 1.16.2
- We only have 1 site per application pool.

```
[w3wp.exe] 7036: [info] Detected environment variable DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION=true
[w3wp.exe] 3636: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 341937672 /LM/W3SVC/1/ROOT/xxxxxxx-1-132345695907704549
[w3wp.exe] 4840: [info] JITCompilationStarted: Startup hook registered in function_id=361337040 token=100688380 name=System.Web.Compilation.BuildManager.InvokePreStartInitMethods(), assembly_name=System.Web app_domain_id=1937404912 domain_neutral=1
[w3wp.exe] 3636: [info] AssemblyLoadFinished: Datadog.Trace.ClrProfiler.Managed v1.17.0 matched profiler version v1.17.0
[w3wp.exe] 3636: [info] AssemblyLoadFinished: Datadog.Trace.ClrProfiler.Managed was not loaded domain-neutral
[w3wp.exe] 3636: [warn] JITCompilationStarted skipping method: Method replacement found but the managed profiler has not yet been loaded into AppDomain with id=1937404912 function_id=648750164 token=100665474 caller_name=System.Data.Linq.SqlClient.SqlProvider.Execute() target_name=System.Data.Common.DbCommand.ExecuteReader()
[w3wp.exe] 3636: [warn] JITCompilationStarted skipping method: Method replacement found but the managed profiler has not yet been loaded into AppDomain with id=1937404912 function_id=648750164 token=100665474 caller_name=System.Data.Linq.SqlClient.SqlProvider.Execute() target_name=System.Data.Common.DbCommand.ExecuteReader()
[w3wp.exe] 3636: [warn] JITCompilationStarted skipping method: Method replacement found but the managed profiler has not yet been loaded into AppDomain with id=1937404912 function_id=648750164 token=100665474 caller_name=System.Data.Linq.SqlClient.SqlProvider.Execute() target_name=System.Data.Common.DbCommand.ExecuteReader()
[w3wp.exe] 3636: [warn] JITCompilationStarted skipping method: Method replacement found but the managed profiler has not yet been loaded into AppDomain with id=1937404912 function_id=648750164 token=100665474 caller_name=System.Data.Linq.SqlClient.SqlProvider.Execute() target_name=System.Data.Common.DbCommand.ExecuteNonQuery()
```

**Comment by lucaspimentel** [comment]
@scownTownV: We recently found out that the flag `DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION` did not cover all intended cases. The fix (#738) should be ready for the next tracer release.

**Comment by scownTownV** [comment]
@lucaspimentel do you have an ETA on the next release date?

**Comment by lucaspimentel** [comment]
@scownTownV: the original fix in #738 did not pan out as hoped, so those changes were reverted in #749. We're actively working on even larger changes (#750) that will fix the root cause and remove the need for the `DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION` workaround.

**Comment by zacharycmontoya** [comment]
Hi @scownTownV , we've just released .NET Tracer 1.18.0 which includes the large, underlying fix Lucas was referring to. Can you please upgrade and let us know if you are still experiencing this issue?

**Comment by scownTownV** [comment]
@zacharycmontoya thanks. Unfortunately this latest version broke our website. Looks like some sort of issue with redis (which we use) but we'll need to look into it further on our side. 


[FileNotFoundException: Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.]    System.ReadOnlySpan`1..ctor(T[] array, Int32 start, Int32 length) +0    StackExchange.Redis.ServerSnapshot.get_Span() in C:\projects\stackexchange-redis\src\StackExchange.Redis\ConnectionMultiplexer.cs:990    StackExchange.Redis.ConnectionMultiplexer.ActivateAllServers(LogProxy log) in C:\projects\stackexchange-redis\src\StackExchange.Redis\ConnectionMultiplexer.cs:1405    StackExchange.Redis.<ReconfigureAsync>d__142.MoveNext() in C:\projects\stackexchange-redis\src\StackExchange.Redis\ConnectionMultiplexer.cs:1714
--

**Comment by lucaspimentel** [comment]
@scownTownV: The issues with instrumenting shared assemblies are resolved in 1.18.0, so we're closing this issue. The new error you're seeing seems similar to #771, so let's move the conversation there. Thanks.
